### PR TITLE
feat: add ability to rename native wallets

### LINKED ISSUE #498: Add ability to rename native wallets
Needs UX/UI

### PR DESCRIPTION
**Pull request by jdambron**
## Description

Add the ability to rename native wallets

- Replaced the _Forget?_ button with a _DeleteIcon_
- Added a _EditIcon_ button for the rename action
- Added a dedicated step in the modal (with new nickname and password to decrypt wallet)
- Handling of the rename action

See video below for visual details.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #498 

## Screenshots (if applicable)

https://user-images.githubusercontent.com/5672954/147755638-eae1e681-1b92-4b2f-a941-bd4c7be6b06e.mov

### CODE REVIEW

File: src/context/WalletProvider/NativeWallet/components/NativeLoad.tsx [hunk 1]
```diff
@@ -153,15 +161,18 @@ export const NativeLoad = () => {
...
+                <IconButton
+                  aria-label={translate('common.rename')}
+                  variant='ghost'
+                  icon={<EditIcon />}
+                  onClick={() => handleRename(wallet)}
+                />
+                <IconButton
+                  aria-label={translate('common.forget')}
+                  variant='ghost'
+                  icon={<DeleteIcon />}
```
Comment (theoboldfrazier) [inline src/context/WalletProvider/NativeWallet/components/NativeLoad.tsx:173]: We'll want to wrap these two icons in a Box to keep them together on the right and prevent them from spacing out when the wallet name is short.

**Review by theoboldfrazier — APPROVED** [review]
Tested out locally and it worked. I left one comment about the UI, but other than that it looks good.

**Comment by jdambron** [comment]
@theoboldfrazier thank you for your feedback.
I wrapped the 2 icons inside a `Box` as suggested.

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by 0xdef1cafe** [comment]
🔥 🇫🇷

**Comment by cjthompson** [comment]
The edit button needs to be on the right next to the delete button not weirdly floating in the middle. Can you make a screenshot where there are multiple wallets with both short and very long names?

What is the character limit on name?

**Comment by cjthompson** [comment]
You don't have any validations for the "name" input. What happens if they input an empty name is given? What if the name is crazy long? is HTML escaped?

**Pull request merged**